diff --git i/CMakeLists.txt w/CMakeLists.txt
index 84d4198b..e72abf05 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -109,10 +109,7 @@ if(MLX_BUILD_METAL)
   endif()
 
   # Throw an error if xcrun not found
-  execute_process(
-    COMMAND zsh "-c" "/usr/bin/xcrun -sdk macosx --show-sdk-version"
-    OUTPUT_VARIABLE MACOS_SDK_VERSION
-    OUTPUT_STRIP_TRAILING_WHITESPACE COMMAND_ERROR_IS_FATAL ANY)
+  set(MACOS_SDK_VERSION @sdkVersion@)
 
   if(${MACOS_SDK_VERSION} LESS 14.0)
     message(

